src/Http/QueryString.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * Copyright 2016 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\SamlBundle\Http;
+
+use Surfnet\SamlBundle\Exception\RuntimeException;
+use Surfnet\SamlBundle\SAML2\AuthnRequest;
+use Symfony\Component\HttpFoundation\Request;
+
+final class QueryString
```

Comment:
Not sure I like this object - actually quite sure I don't. It accepts everything (even requests without a querystring, or non-GET requests) and verifies nothing (the presence of the required parameters for the getSignedRequestQuery, whether or not a query is even present etc). Also, to me, methodnames are a bit weird:
- `countParameter`, what does that count, the parameters, the occurrences of a certain parameter, or the amount of occurrences with values?)
- `getSignedRequestQuery` - it returns, if possibly available since that has not been verified, a "signable" request query, that may or may not have been signed, which is not verified

Furthermore why is the signature itself not retrieved from this object?

To me it seems there was a good idea behind it, but it doesn't really live up to expectation.

I.m.h.o. Either, use this, use it well and ensure that this is the single source of truth for the request query of the current AuthnREquest (which checks whether or not the query is structurally valid etc) as a value object, or do not use it at all.